packages/tangle/cmanaweightprovider.go

```diff
@@ -88,14 +99,19 @@ func (c *CManaWeightProvider) WeightsOfRelevantSupporters() (weights map[identit
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
 
-	for nodeID, t := range c.activeNodes {
-		if targetTime.Sub(t) > activeTimeThreshold {
-			delete(c.activeNodes, nodeID)
+	for nodeID, al := range c.activeNodes {
+		nodeMana := mana[nodeID]
+
+		// Skip node if it does not fulfill minimumManaThreshold.
+		if nodeMana <= minimumManaThreshold {
 			continue
 		}
 
-		nodeMana := mana[nodeID]
-		if nodeMana <= minimumManaThreshold {
+		// Determine whether node was active in time window.
+		if active, empty := al.Active(targetTime.Add(-activeTimeThreshold), targetTime); !active {
```

Comment:
Extract `targetTime.Add(-activeTimeThreshold)` to a var before the `for` loop as it is going to be the same for each iteration.